RyleyGG/apluschiever
# 14 15

# Issue
Closes #14, #15 

# Description
This PR implements some basic filter objects and endpoints for grabbing user and course data. These will of course necessarily expand over time, but the framework has been implemented. A large refactor was also implemented to use SQLModel for interacting with our database. SQLModel sits on top of the SQLAlchemy ORM and Pydantic library and will significantly reduce duplicate code and development friction.

# Type Of Change

# Test Steps
1. Stand up application
2. Ensure webpage runs without errors
3. Hit endpoints via Postman
4. Run pytest framework

# Checklist
